geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSCardIntegrationTest.java

```diff
@@ -0,0 +1,106 @@
+import static org.apache.geode.redis.internal.RedisConstants.ERROR_WRONG_TYPE;
+import static org.apache.geode.test.dunit.rules.RedisClusterStartupRule.BIND_ADDRESS;
+import static org.apache.geode.test.dunit.rules.RedisClusterStartupRule.REDIS_CLIENT_TIMEOUT;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import redis.clients.jedis.HostAndPort;
+import redis.clients.jedis.JedisCluster;
+import redis.clients.jedis.Protocol;
+
+import org.apache.geode.redis.ConcurrentLoopingThreads;
+import org.apache.geode.redis.RedisIntegrationTest;
+
+public abstract class AbstractSCardIntegrationTest implements RedisIntegrationTest {
+  private JedisCluster jedis;
+  private static final String key = "{user1}setKey";
+
+  @Before
+  public void setUp() {
+    jedis = new JedisCluster(new HostAndPort(BIND_ADDRESS, getPort()), REDIS_CLIENT_TIMEOUT);
+  }
+
+  @After
+  public void tearDown() {
+    flushAll();
+    jedis.close();
+  }
+
+  @Test
+  public void scardWrongNumOfArgs_returnsError() {
+    assertExactNumberOfArgs(jedis, Protocol.Command.SCARD, 1);
+  }
+
+  @Test
+  public void scardSet_returnsSize() {
+    String[] values = new String[] {"Jeff", "Natalie", "Michelle", "Joe", "Kelley"};
+    jedis.sadd(key, values);
+    assertThat(jedis.scard(key)).isEqualTo(values.length);
+  }
+
+  @Test
+  public void scardNonExistentSet_returnsZero() {
+    assertThat(jedis.scard("{user1}nonExistentSet")).isEqualTo(0);
+  }
+
+  @Test
+  public void scardAfterSadd_returnsSize() {
+    String[] valuesInitial = new String[] {"Jeff", "Natalie", "Michelle", "Joe", "Kelley"};
+    String[] valuesToAdd = new String[] {"one", "two", "three"};
+
+    jedis.sadd(key, valuesInitial);
+    long size = jedis.scard(key);
+    assertThat(size).isEqualTo(valuesInitial.length);
+    jedis.sadd(key, valuesToAdd);
+    assertThat(jedis.scard(key)).isEqualTo(size + valuesToAdd.length);
+  }
+
+
+  @Test
+  public void scardConcurrentSAddSCard_sameKeyPerClient() {
+    String[] valuesInitial = new String[] {"one", "two", "three"};
+    String[] valuesToAdd = new String[] {"pear", "apple", "plum", "orange", "peach"};
+    jedis.sadd(key, valuesInitial);
+
+    final AtomicLong scardReference = new AtomicLong();
+    new ConcurrentLoopingThreads(1000,
+        i -> jedis.sadd(key, valuesToAdd),
+        i -> scardReference.set(jedis.scard(key)))
+            .runWithAction(() -> {
+              assertThat(scardReference).satisfiesAnyOf(
+                  scardResult -> assertThat(scardResult.get()).isEqualTo(valuesInitial.length),
+                  scardResult -> assertThat(scardResult.get())
+                      .isEqualTo(valuesInitial.length + valuesToAdd.length));
+              jedis.srem(key, valuesToAdd);
+              jedis.sadd(key, valuesToAdd);
```

Comment:
This sadd should be removed, as with it in place, the sadd in the ConcurrentLoopingThreads won't change the size of the set, as the elements it's adding will always already be present. If that's the case, then there will be no chance of a concurrency issue between that sadd and scard.